developer-docs/bolt_server.md

```diff
@@ -0,0 +1,145 @@
+- `sudo-password`: String, *optional* - Password to use when changing users via `run-as`
+
+### Task Object
+The Task be a JSON object. The following keys are available:
+- `name`: String, *required* - The name of the task.
+- `metadata`: String, *optional* - The contents of the task's metadata.json file. 
+- `file`: String, *required* - Task's base64 encoded file content.
+
+## Response 
+If the task runs the response will have status 200.
+The response will be a standard bolt Result JSON object.
+
+## Setup for manual testing changes
+
+The following example walks through the setup to get the bolt-server running on an Ubuntu 16.04 vmpooler node. 
+
+1. Download and install puppet6 and puppetserver (for certs). The latest release can be found at http://nightlies.puppet.com/
+```
+curl -O http://nightlies.puppet.com/apt/puppet6-nightly-release-xenial.deb
+dpkg -i ./puppet6-nightly-release-xenial.deb 
+apt-get update
+apt-get install puppetserver
+export PATH=$PATH:/opt/puppetlabs/bin
+puppet resource service puppetserver ensure=running
+```
+2. Get certs (Error messages OK here, as long as certs are generated)
+```
+puppet agent -t
+puppet cert list -a
+puppet cert sign [host reported from command above]
+```
+Check that the following certs have been generated
+```
+/etc/puppetlabs/puppet/ssl/certs/$HOSTNAME.pem
+/etc/puppetlabs/puppet/ssl/private_keys/$HOSTNAME.pem
+```
+3. Download and install bolt-server. The latest release can be found at http://builds.delivery.puppetlabs.net/pe-bolt-server
+```
+curl -O http://builds.delivery.puppetlabs.net/pe-bolt-server/0.21.7/artifacts/deb/xenial/pe-bolt-server_0.21.7-1xenial_amd64.deb
+dpkg -i pe-bolt-server_0.21.7-1xenial_amd64.deb
+```
+4. Copy over certs to bolt-server directory
+
+- `cp /etc/puppetlabs/puppet/ssl/certs/$HOSTNAME.pem /etc/puppetlabs/bolt-server/ssl/$HOSTNAME.cert.pem`
+- `cp /etc/puppetlabs/puppet/ssl/private_keys/$HOSTNAME.pem /etc/puppetlabs/bolt-server/ssl/$HOSTNAME.key.pem`
+- `cp /etc/puppetlabs/puppet/ssl/certs/ca.pem /etc/puppetlabs/bolt-server/ssl/ca.pem`
+5. Write config file
+Save following as `/etc/puppetlabs/bolt-server/conf.d/bolt-server.conf`
+```
+bolt-server: {
+  ssl-cert: "/etc/puppetlabs/bolt-server/ssl/$HOSTNAME.cert.pem"
+  ssl-key: "/etc/puppetlabs/bolt-server/ssl/$HOSTNAME.key.pem"
+  ssl-ca-cert: "/etc/puppetlabs/bolt-server/ssl/ca.pem"
+}
+```
+6. Grant permissions to bolt-server
+
+`chown -R pe-bolt-server:pe-bolt-server /etc/puppetlabs/bolt-server/*`
+
+7. Start bolt-server
+`service pe-bolt-server start`
+8. Build a request
+Save the following JSON to `~/request.json`
+```
+{
+  "target": {
+  "hostname": "juvct8xnyr2ihvm.delivery.puppetlabs.net",
+  "user": "root",
+  "password": "Secret",
+  "host-key-check": "false"
+  },
+  "task": {
+  "name": "echo",
+  "metadata": {
+    "description": "Echo a message",
+    "parameters": {
+    "message": "Default string"
+    }
+        },
```

Comment:
Some weird indentation :P

Reply:
Yeah, this was copied from google doc to vi on test VM, then to markdown. Verified it worked, not that it was readable or made any sense. Updated.